apps/ledger-live-desktop/src/newArch/hooks/useCopyToClipboard.ts

```diff
@@ -0,0 +1,9 @@
+import { useCallback } from "react";
+
+export function useCopyToClipboard() {
```

Comment:
I still haven't tested yet (I'll do it tomorrow morning). But in the mean time I noticed in my PR that tapping on the copy buttons but not getting any feedback is not great it feels like nothing actually happened. This is why the legacy buttons show a tick mark after clicking. Although it does not look very good it makes for a much better UX IMO.

As a result I implemented the LLM hook like this:
```ts
export function useCopyToClipboard(callback?: (text: string) => void) {
  const onCopyRef = useRef<() => void>();
  return (text: string) => {
    if (!onCopyRef.current) {
      onCopyRef.current = () => {
        Clipboard.setString(text);
        callback?.(text);
      };
    }
    return onCopyRef.current;
  };
}
```
This way a callback can be passed to the hook to trigger an animation or any kind of feedback.

The other difference is that it keeps the same reference for the returned function instead for the factory (the returned function is passed down the components but the factory is not) if the returned function reference changes because it's re-declared: memoize components will still re-render (so the `useCallback` on the factory is not useful).

Reply:
Yep definitely a good idea

Reply:
A check icon has been added and some hover and active bg also